.github/workflows/terraform-tests.yml

```
@@ -56,6 +68,8 @@ jobs:
       uses: actions/checkout@v3
       with:
         repository: hashicorp/terraform-provider-aws
+        ref: v4.31.0
+#        ref: ${{ github.event.inputs.terraform-ref }}
```

This seems to be a leftover.
A lot of the tests are currently failing due to changes in the tests (they check for a new attribute in EC2 VPCs which is not yet supported by LocalStack). Therefore using a pinned version would be a good idea for now. But the input should still be used. So the default behavior should be that the terraform-ref input is used when set, but it defaults to the latest ref which is known to be working (i.e. the pinned version).

```suggestion
        ref: ${{ github.event.inputs.terraform-ref || "v4.31.0"}}
```